src/lib/migrations/migrations.ts

```diff
@@ -0,0 +1,116 @@
+import { client, collections } from "$lib/server/database";
+import { migrations } from "./routines";
+import { acquireLock, releaseLock, isDBLocked, refreshLock } from "./lock";
+import { isHuggingChat } from "$lib/utils/isHuggingChat";
+
+export async function checkAndRunMigrations() {
+	// make sure all GUIDs are unique
+	if (new Set(migrations.map((m) => m._id.toString())).size !== migrations.length) {
+		throw new Error("Duplicate migration GUIDs found.");
+	}
+
+	// check if all migrations have already been run
+	const migrationResults = await collections.migrationResults.find().toArray();
+
+	// if all the migrations._id are in the migrationResults, we can exit early
+	if (
+		migrations.every((m) => migrationResults.some((m2) => m2._id.toString() === m._id.toString()))
+	) {
+		console.log("[MIGRATIONS] All migrations already applied.");
```

Comment:
maybe there is no need for this statement

Reply:
I see your point, ideally we would have log levels because this is a bit verbose, but when people need support for chat-ui I think it'd be nice to know if the migrations are working correctly in the logs. :eyes: 

Maybe we can have log levels in the future like debug/info/warning ?

Reply:
perfect!